### DISCUSSION TITLE
Change between different models in a multimodel PDB file without having to reload the widget

### DISCUSSION THREAD

**Discussion by mverissi** [nglviewer/nglview#1059] [Q&A]
Dear developers,

I have a multimodel pdb file generated from a geometry optimization of a molecule and I would like to visualize it using nglview much in the same way as one can do in PyMol. In PyMol, once a multimodel PDB file is loaded, you can switch between the different models through buttons that appear in the lower left of the PyMol window. Is there any way of doing this with nglview without generating a trajectory file such as Gromacs' xtc?

With the multimodel file name already set in the variable `multi_pdb_file`, I tried

```
import nglview as nv

view = nv.show_structure_file(multi_pdb_file, gui=True, default=True)
view.center()
view._remote_call('setSize', target='Widget', args=['','600px'])

view
```

but there is no way to switch, for example, from model 1 to model 2 apart from adding code that explicitly loads that model, in which case the widget has to be reloaded. I would like to avoid this because this distracts me from seeing the (small) changes in geometry from one model to the next (or previous).

Best,

Marcos

**Comment by hainm** [comment]
@mverissi Please try this and let me know if that helps

https://github.com/nglviewer/nglview/issues/765#issuecomment-461952303

**Reply by mverissi** [reply]
Thanks, @hainm . I'll try it and let you know.

**Reply by mverissi** [reply]
Hey @hainm, thanks for the reply. What you suggested works, but it's not what I would like. Your suggestion either displays only one or all structures in the pdb file at a time. However, if I want to switch to another of the structures in the multimodel pdb file, I have to reload the widget. I'm attaching a screenshot of PyMol - in the lower right corner I have circled the animation controls in yellow. It allows me to smoothly switch between adjacent models and have an animation of the geometry optimization.

As far as I can see, the only way I can view an animated trajectory in nglview is to create a Gromacs xtc or trr, or the equivalent files for other MD software, and then use the corresponding method (`nglview.show_mdtraj` for a Gromacs trajectory, `nglview.show_parmed` for a pytraj trajectory, and so on). Is that correct? The problem is that my multimodel pdb is not generated by a molecular dynamics program, so I just have the pdb.

Best,

Marcos

<img width="1470" alt="Screenshot 2023-05-17 at 11 07 11" src="https://github.com/nglviewer/nglview/assets/20373057/a6da1a2d-2fa2-42d0-8e26-85c4ae6a42fc">

**Reply by hainm** [reply]
@mverissi Please see below (I paste the code too)

<img width="919" alt="image" src="https://github.com/nglviewer/nglview/assets/4451957/e7a1f7fc-b6cd-41c6-bd23-adfe8ef0d05e">

```
from ipywidgets import IntSlider
from ipywidgets import link

slider = IntSlider(min=1, max=10)

def on_change(change):
    frame = change['new']
    print(frame)
    view._remote_call('setSelection', target='compList', args=[f'/{frame}'], 
               kwargs=dict(component_index=0))
    
slider.observe(on_change, names='value')
slider
```
